config/config.go

```diff
@@ -723,6 +723,29 @@ func (c *RelabelConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 // Regexp encapsulates a regexp.Regexp and makes it YAML marshallable.
 type Regexp struct {
 	regexp.Regexp
+	original string
+}
+
+// NewRegexp creates a new anchored Regexp and returns an error if the
+// passed-in regular expression does not compile.
+func NewRegexp(s string) (*Regexp, error) {
```

Comment:
Would you then duplicate this constructor in both packages where it's used?
But NewRegexp is also used in the non-test code, and my intention was to
just additionally use it from tests (via Must...) to make sure it's
working. Or did you mean to keep only the Must... variant local?
On Sep 1, 2015 16:05, "Fabian Reinartz" <notifications@github.com> wrote:

> In config/config.go
> <https://github.com/prometheus/prometheus/pull/1044#discussion_r38421787>:
>
> > @@ -723,6 +723,29 @@ func (c *RelabelConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
> >  // Regexp encapsulates a regexp.Regexp and makes it YAML marshallable.
> >  type Regexp struct {
> >  	regexp.Regexp
> > +	original string
> > +}
> > +
> > +// NewRegexp creates a new anchored Regexp and returns an error if the
> > +// passed-in regular expression does not compile.
> > +func NewRegexp(s string) (*Regexp, error) {
>
> In other places we usually just make such constructors local the tests if
> they are necessary to setup table tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/prometheus/prometheus/pull/1044/files#r38421787>.
>